packages/pn-commons/src/utils/notification.utility.ts

```diff
@@ -897,6 +911,33 @@ function timelineElementMustBeShown(t: INotificationDetailTimeline): boolean {
   return TimelineAllowedStatus.includes(t.category);
 }
 
+/**
+ * Insert cancelled status n timeline.
+ * @param  {NotificationDetail} notificationDetail
+ */
+const insertCancelledStatusInTimeline = (notificationDetail: NotificationDetail) => {
+  const timelineCancelledElement = notificationDetail.timeline.find(
+    (el) => el.category === TimelineCategory.NOTIFICATION_CANCELLED
+  );
+  if (!timelineCancelledElement) {
+    notificationDetail.timeline.forEach((element) => {
```

Comment:
the forEach is useless

Reply:
if i don'use for each, where i take element?

Reply:
you are taking with the find

Reply:
do it